pdns/dnspacket.cc

```diff
@@ -756,3 +759,14 @@ bool DNSPacket::checkForCorrectTSIG(UeberBackend* B, DNSName* keyname, string* s
 const DNSName& DNSPacket::getTSIGKeyname() const {
   return d_tsigkeyname;
 }
+
+#ifdef ENABLE_GSS_TSIG
+void DNSPacket::cleanupGSS(int rcode)
+{
+  if (rcode != RCode::NoError && d_tsig_algo == TSIG_GSS && !getTSIGKeyname().empty()) {
```

Comment:
should this also check if the feature is enabled globally? The downstream `erase` is, of course, very cheap, so perhaps it's fine like this.

Reply:
I will review all the spots where checking `g_doGssTSIG` might be relevant.